packages/amplify-cli-core/src/category-interfaces/amplifyBaseCDKTypes.ts

```diff
@@ -0,0 +1,35 @@
+//Define all common classes and interfaces required to generate cloudformation using CDK.
+import * as cdk from '@aws-cdk/core';
+
+//Base template
+//Customer can use these params to mutate the Cloudformation for the resource
+export interface AmplifyStackTemplate {
+  addCfnParameter(props: cdk.CfnParameterProps, logicalId: string): void;
+  addCfnOutput(props: cdk.CfnOutputProps, logicalId: string): void;
+  addCfnMapping(props: cdk.CfnMappingProps, logicalId: string): void;
+  addCfnCondition(props: cdk.CfnConditionProps, logicalId: string): void;
+
+  getCfnParameter(logicalId: string): cdk.CfnParameter;
+  getCfnOutput(logicalId: string): cdk.CfnOutput;
+  getCfnMapping(logicalId: string): cdk.CfnMapping;
+  getCfnCondition(logicalId: string): cdk.CfnCondition;
+}
+
+/**
+ * Temporary type for generated CFN Template
+ */
+export interface Template {
```

Comment:
please rename this to declare the exact use of the structure. e.g. AmplifyGeneratedCFNTemplate or something similar. Template is an overloaded term.

Reply:
Cloudform also defined as Template , thats why I named as Template